src/plugins/intel_gna/gna_device.cpp

```diff
@@ -444,6 +457,10 @@ GnaWaitStatus GNADeviceHelper::wait(uint32_t reqId, int64_t millisTimeout) {
     }
     checkGna2Status(status, "Gna2RequestWait");
 
+    if (debugLogEnabled) {
+        dumpAllAllocations(debugLogIndexRequestWait, "AfterGna2RequestWait");
+        debugLogIndexRequestWait++;
```

Comment:
```
if( debugLogIndexRequestWait >=std::numeric_limits<uint64_t>::max())
{
     debugLogIndexRequestWait = 0;
}
debugLogIndexRequestWait++;
```

This comment can be omitted due the fact the value range is quite big  and this is for debugging.

Reply:
IMO it is not needed not only UINT64_MAX is practically unreachable but also...

AFAIK "A computation involving unsigned operands can never overﬂow, because a result that cannot be represented by the resulting unsigned integer type is reduced modulo the number that is one greater than the largest value that can be represented by the resulting type."

see
https://stackoverflow.com/questions/18195715/why-is-unsigned-integer-overflow-defined-behavior-but-signed-integer-overflow-is

Am I missing something?

Reply:
your are right for unsigned it is redundant.
  based on documentation https://en.cppreference.com/w/cpp/language/operator_arithmetic

Overflows
```
Unsigned integer arithmetic is always performed modulo 2n
 where n is the number of bits in that particular integer. E.g. for unsigned int, adding one to [UINT_MAX](https://en.cppreference.com/w/cpp/types/climits) gives ​0​, and subtracting one from ​0​ gives [UINT_MAX](https://en.cppreference.com/w/cpp/types/climits).
```